In [1]:
import h5py

Let's load the entire file.

In [7]:
f = h5py.File("/data/genome-attention/l131k_w128.h5", 'r')

In [8]:
list(f.keys())

['pool_width',
 'target_labels',
 'test_in',
 'test_na',
 'test_out',
 'train_in',
 'train_na',
 'train_out',
 'valid_in',
 'valid_na',
 'valid_out']

In [9]:
f['target_labels']

<HDF5 dataset "target_labels": shape (4229,), type "|S29">

**Training In**

Size: 13,047 

Length: 131,072 bp

Shape: 131,072 x 4

In [10]:
f['train_in']

<HDF5 dataset "train_in": shape (13047, 131072, 4), type "|b1">

**Training Out**

Size: 13,047 

Length: 1024 buckets 

Labels: 4229 labels

Shape: 1024 x 4229


In [13]:
f['train_out'][0]

array([[1.034 , 0.    , 0.9663, ..., 0.    , 0.    , 0.4238],
       [0.4507, 0.    , 0.    , ..., 0.    , 0.    , 0.    ],
       [0.7114, 0.    , 0.3726, ..., 0.    , 0.    , 0.    ],
       ...,
       [1.17  , 0.2146, 1.964 , ..., 0.    , 0.    , 0.    ],
       [1.674 , 0.0792, 0.9824, ..., 0.    , 0.    , 0.    ],
       [1.825 , 0.    , 0.0983, ..., 0.    , 0.    , 0.    ]],
      dtype=float16)